From 9fe1b4a8f9eae290996d554458fb7d6e63f2eaae Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Mon, 23 Jan 2023 18:12:57 +0200
Subject: [PATCH backport 5.10 181/181] hwmon: (mlxreg-fan) TMP: Do not return
 negative value for fan speed configuration setting

Do not return negative return value for fan speed configuration
request.

Downstream patch - should be removed when new thermal algorithm is
available.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/hwmon/mlxreg-fan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/mlxreg-fan.c b/drivers/hwmon/mlxreg-fan.c
index 4a8becdb0..acba9d688 100644
--- a/drivers/hwmon/mlxreg-fan.c
+++ b/drivers/hwmon/mlxreg-fan.c
@@ -358,7 +358,7 @@ static int mlxreg_fan_set_cur_state(struct thermal_cooling_device *cdev,
 		 * For configuration non-zero value is to be returned to avoid thermal
 		 * statistics update.
 		 */
-		config = 1;
+		config = 0; /*1*/;
 		state -= MLXREG_FAN_MAX_STATE;
 		for (i = 0; i < state; i++)
 			pwm->cooling_levels[i] = state;
-- 
2.20.1

